cosmoz-omnitable-treenode-column.html

```diff
@@ -46,347 +44,346 @@
+					*/
+					bindValues: {
+						type: Boolean,
+						readOnly: true,
+						value: true
+					},
+		
+					filter: {
+						type: String,
+						value: null
+					},
+		
+					locale: {
+						type: String,
+						value: null
+					},
+		
+					showMaxNodes: {
+						type: Number
+					},
+		
+					/**
+					* If displayed node share a common path from root, this property indicates wether this common
+					* path should be replaced by an ellipsis.
+					*/
+					showCommonPath: {
+						type: Boolean,
+						value: false
+					},
+		
+					/**
+					* Name of the property used to lookup the displayed node in the tree
+					*/
+					keyProperty: {
+						type: String,
+					},
+		
+					valueProperty: {
+						type: String,
+						value: 'name'
+					},
+		
+					/**
+					* The value of the `paper-autocomplete` input.
+					*/
+					query: {
+						type: String
+					},
+		
+					_collator: {
+						type: Object,
+						computed: '_computeCollator(locale)'
+					},
+		
+					_hideFromRoot: {
+						type: Number,
+						value: 0,
+						computed: '_computeHideFromRoot(showCommonPath, values, ownerTree)'
+					},
+		
+					_tooltip: {
+						type: String,
+						computed: '_computeTooltip(filter)'
+					},
+		
+					_showResultsOnFocus: {
+						type: Boolean,
+						value: true,
+						computed: '_computeShowResultsOnFocus(filter)'
+					},
+		
+					minWidth: {
+						type: String,
+						value: '85px'
+					},
+		
+					editMinWidth: {
+						type: String,
+						value: '85px'
+					},
+		
+					/* eslint no-invalid-this: 0 */
+		
+					/**
+					* Function used to filter available items.
+					* This function is actually used by `paper-autocomplete`,
+					* it is also exposed here so it is possible to provide a custom queryFn.
+					*/
+					queryFn: {
+						type: Function,
+						value: () => {
+							return function (datasource, query) {
+								const notNull = i => i != null;
+								return datasource.filter(notNull)
+									.map(item => {
+										let text,
+											value;
+		
+										if (typeof item === 'object') {
+											text = item[this.textProperty];
+											value = item[this.valueProperty];
+										} else {
+											value = text = item.toString();
+										}
+		
+										if (text == null) {
+											return;
+										}
+		
+										// if there is text search for indexOf query
+										if (query == null) {
+											return {
+												text,
+												value,
+												html: text
+											};
+										}
+		
+										if (text.toLowerCase().indexOf(query) < 0) {
+											return;
+										}
+		
+										// Highlight matches
+										return {
+											text,
+											value,
+											html: text.replace(new RegExp('(' + query + ')', 'ig'), '<b>$1</b>')
+										};
+		
+									}).filter(notNull);
+							};
+						}
+					}
+				};
+			}
+			static get observers() {
+				return [
+					'_setQueryByPath(filter, ownerTree)'
+				];
+			}
+			constructor() {
```

Comment:
If the only thing we're doing in the constructor is calling super, do we even need a constructor?

Reply:
no, no we do not